vminitd/Sources/vminitd/ManagedContainer.swift

```diff
@@ -28,6 +36,370 @@ actor ManagedContainer {
     private let log: Logger
     private let bundle: ContainerizationOCI.Bundle
     private var execs: [String: ManagedProcess] = [:]
+    private var namespaceWorker: NamespaceWorker?
+
+    /// Worker child process that runs in container's namespace for filesystem operations
+    private final class NamespaceWorker: @unchecked Sendable {
```

Comment:
A few things here:

- Is there a reason for nesting this class? It's adding about 350 lines to ManagedContainer, which makes it harder for someone unfamiliar with the code to reason about either. I can move it outside of ManagedContainer and still compile it, so let's move it into its own file.
- I could see this being called NamespaceWorker if it facilitated any kind of work taking place between the guest agent and namespace, which could be a useful abstraction later on. For now let's change the name to describe specifically the work being done.
- We need to manage the concurrency between guest agent, worker parent, and worker child such that `@unchecked` isn't needed for Sendable. In particular, we can't do blocking socket `read()` calls in Task context.